jwt.js

```diff
@@ -39,7 +39,13 @@ function fastifyJwt (fastify, options, next) {
   let secretOrPrivateKey
   let secretOrPublicKey
 
-  if (typeof secret === 'object' && !Buffer.isBuffer(secret)) {
+  if (typeof secret === 'function' && secret.then === 'function') {
```

Comment:
```suggestion
  if (typeof secret === 'function' && typeof secret.then === 'function') {
```
Missing `typeof` here